src/journal/journald-context.c

```diff
@@ -784,3 +840,40 @@ void client_context_acquire_default(Server *s) {
 
         }
 }
+
+#if HAVE_PCRE2
+static bool is_log_matched(pcre2_code *regex, const char *message) {
```

Comment:
This feels generic enough that it could go into a function `pattern_matches()` in prcre2-util.h instead of being defined here.

Reply:
Return type here should be int since log_oom() returns -ENOMEM